projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts

```diff
@@ -210,30 +220,38 @@ export class PerformanceViewComponent implements OnInit, OnDestroy {
     if (this.includeThreadGroupChart) {
       this.createThreadGroupsChart(this.findRequest);
     }
-  }
+    this.initializationTasks.push(this.tableInitializedSubject); // we wait also for the table
+    this.initializationTasks.push(this.rangerLoadedSubject);
 
-  // we assume the settings input has been changed
-  reconstructAllCharts() {
-    this.findRequest = this.prepareFindRequest(this.settings); // we don't want to lose active filters
-    this.updateAllCharts();
-  }
-
-  mergeRequestWithActiveFilters() {
-    this.deleteObjectProperties(this.findRequest.params, this.filtersComponent.getAllFilterAttributes()); // we clean the request first
-    Object.assign(this.findRequest.params, this.executionContext.getActiveFilters());
+    forkJoin(this.initializationTasks.map((obs) => obs.pipe(take(1)))).subscribe((allCompleted) =>
+      this.onInitializationComplete.emit()
+    );
   }
 
   updateAllCharts() {
+    this.updateTasks = [];
```

Comment:
Do we really need the differentiation of updateTasks and initializeTasks? I see a lot of duplications between these.